pkg/controller/utils/immutable_image.go

```diff
@@ -0,0 +1,41 @@
+package utils
+
+import (
+	"fmt"
+	"github.com/Dynatrace/dynatrace-oneagent-operator/pkg/apis/dynatrace/v1alpha1"
+	"github.com/Dynatrace/dynatrace-oneagent-operator/pkg/dtclient"
+	"github.com/Dynatrace/dynatrace-oneagent-operator/pkg/version"
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"time"
+)
+
+const updateInterval = 5 * time.Minute
+
+// SetUseImmutableImageStatus sets the UseImmutableImage and LastClusterVersionProbeTimestamp stati of an BaseOneAgentDaemonSet instance
+// Returns true if:
+//     UseImmutableImage of specification is true &&
+//			LastClusterVersionProbeTimestamp status is the duration of updateInterval behind
+// otherwise returns false
+func SetUseImmutableImageStatus(logger logr.Logger, instance v1alpha1.BaseOneAgent, agentVersion string, dtc dtclient.Client) bool {
+	if dtc == nil {
+		err := fmt.Errorf("dynatrace client is nil")
+		logger.Error(err, err.Error())
+		return false
+	}
+
+	// Variable declared to make if-condition more readable
+	lastClusterVersionProbeTimestamp := instance.GetStatus().LastClusterVersionProbeTimestamp.UTC()
+	if instance.GetSpec().UseImmutableImage && isLastProbeOutdated(lastClusterVersionProbeTimestamp) {
+		instance.GetStatus().LastClusterVersionProbeTimestamp = metav1.Now()
+		instance.GetStatus().UseImmutableImage =
+			version.IsRemoteClusterVersionSupported(logger, dtc) &&
+				version.IsAgentVersionSupported(logger, agentVersion)
```

Comment:
There is also the OneAgent version configured on the cluster.

For example, customer installing Operator in a new environment using our instructions and we have `.spec.useImmutableImage` enabled by default in our example, their cluster supports images, but the OneAgent version they want to use (and **it's configured on Dynatrace**, i.e., not on .spec) doesn't, we need to disable immutable image for this case.

In fact the version on `.spec` is not particularly relevant since we don't modify the installer URL anyway if the version is unsupported.

This may not be an uncommon situation at the beginning once we release this feature.